src/openzaak/components/documenten/management/commands/delete_empty_eioc.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2024 Dimpact
+from typing import List
+
+from django.conf import settings
+from django.core.management import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+
+from openzaak.components.documenten.models import EnkelvoudigInformatieObjectCanonical
+
+
+class Command(BaseCommand):
+    help = "Check for any empty document identities (no related document)."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--noinput",
+            "--no-input",
+            action="store_false",
+            dest="interactive",
+            help=_("Delete automatically any empties"),
+        )
+
+    def handle(self, *args, **options):
+        if settings.CMIS_ENABLED:
+            self.stdout.write(_("This command does not run with CMIS enabled."))
+            return
+
+        msg = _("Checking {count} records ...").format(
+            count=EnkelvoudigInformatieObjectCanonical.objects.count()
+        )
+        self.stdout.write(msg)
+
+        duplicates = EnkelvoudigInformatieObjectCanonical.objects.filter(
+            enkelvoudiginformatieobject__isnull=True
+        )
+
+        if len(duplicates) == 0:
+            self.stdout.write(_("Found no empty records."))
+            return
+
+        msg = _("Found {count} empty records.").format(count=len(duplicates))
+        self.stdout.write(msg)
+
+        if not options["interactive"]:
+            delete_empties(duplicates)
+            self.stdout.write(self.style.SUCCESS(_("Deleted all the empties.")))
+            return
+
+        while True:
+            option = self.get_option()
```

Comment:
Looks good, but let's make it less interactive :) 
The idea is that this command can be used in some deployment script without user input. 

Options look nice, so maybe add `interactive` flag to the command if you want to keep the them? 

And could you please create a regression test with this command, which is based on this issue with following steps:
* create canonical object without latest_version
* relate zaak to it
* try to patch the zaak (it should fail)
* call the command in the non-interactive mode
* try to patch the zaak again (success)

Reply:
I've just removed the interactivity